src/Scripts/Game/Features/Processing/EL_ProcessAction.c

```diff
@@ -0,0 +1,63 @@
+class EL_ProcessAction : ScriptedUserAction
+{
+	[Attribute(defvalue: "", uiwidget: UIWidgets.ResourcePickerThumbnail, desc: "Prefab to Input", "et")]
+	ResourceName m_PrefabToInput;
+	
+	[Attribute(defvalue: "1", uiwidget: UIWidgets.Auto, desc: "Input amount per process")]
+	private int m_InputAmount;
+	
+	[Attribute(defvalue: "", uiwidget: UIWidgets.ResourcePickerThumbnail, desc: "Prefab to Output", "et")]
+	ResourceName m_PrefabToOutput;		
+	
+	[Attribute(defvalue: "1", uiwidget: UIWidgets.Auto, desc: "Output amount per process")]
+	private int m_OutputAmount;
+	
+	//------------------------------------------------------------------------------------------------
+	override void PerformAction(IEntity pOwnerEntity, IEntity pUserEntity)
+	{
+		InventoryStorageManagerComponent inventoryManager = InventoryStorageManagerComponent.Cast(pUserEntity.FindComponent(SCR_InventoryStorageManagerComponent));
+
+		array<IEntity> allFoundItems = GetAllInputItems(inventoryManager);
+		
+		for (int i = 0; i < m_InputAmount; i++) 
+		{
+			inventoryManager.TryDeleteItem(allFoundItems[i]);
+		}
```

Comment:
Should do double check if we were able to remove the amount and if `TryDeleteItem` fails and abort the action? In that case, the items already deleted would be lost though. Not sure how reliable CanPerform vs Perform is and if users could do any funky tricks like removing items from inventory while processing or something.

Reply:
All items used are searched in `PerformAction`.
Added another check in `PerformAction` for all input items, so even if someone removes the items while performing the action it would just do nothing.